### DISCUSSION TITLE
is it broken, or am I?

### DISCUSSION THREAD

**Discussion by amazingDJname** [tonaljs/tonal#422] [Q&A] [answered]
Total js noob here, I was watching this delicious max tutorials video : https://www.youtube.com/watch?v=RR5AlDgYI2s

and when I get the code and try it on my end, the detect function gives me totally wrong chords - when i play a c major triad for example, it will return G64 Em#5. When I put in a second inversion C major chord, it will return E64 Cm#5

Anyone else ever encounter this issue, or could help point me in the right direction?

const maxApi = require("max-api");
const tonal = require("tonal");
const Detect = require("tonal-detect");

maxApi.outlet("Hello you fancy lad");

maxApi.addHandler("challenge", () => {
	maxApi.outlet("Unhand my blancmange you swine!");
});

maxApi.addHandler("detect", (...midiNotes) => {
	const namedNotes = midiNotes.map(tonal.Note.fromMidi);
	const chords = Detect.chord(namedNotes);
	if (chords && chords.length > 0) {
		maxApi.outlet("chords", ...chords);
	} else {
		maxApi.outlet("chords", "unknown");
	}
});

**Comment by danigb** [comment, marked as answer]
Hey:

As I mentioned several times, chord detection is not very well polished inside the library. Anyway, I think you're using an older version. You can use `Chord.detect` directly without need of loading another library.

Those are the results I get on latest:
```js
import { Chord } from 'tonal';
Chord.detect(["c", "e", "g"]) // => [ 'CM', 'Em#5/C' ]
Chord.detect(["e", "g", "c"]) // => [ 'Em#5', 'CM/E' ]
```

**Reply by amazingDJname** [reply]
solved! thank you boss!
